subprojects/platform-play/src/main/java/org/gradle/play/internal/run/DefaultVersionedPlayRunAdapter.java

```diff
@@ -145,24 +142,41 @@ private void closeOldLoaders() throws IOException {
     public void buildSuccess() {
         reload.set(true);
         buildFailure = null;
-        makeReady(true);
+        pendingChanges.done();
     }
 
     @Override
     public void buildError(Throwable throwable) {
         buildFailure = throwable;
-        makeReady(true);
+        pendingChanges.done();
     }
 
     @Override
-    public void rebuildInProgress() {
-        makeReady(false);
+    public void expectPendingChanges() {
+        pendingChanges.more();
+
     }
 
-    private void makeReady(boolean isReady) {
-        synchronized (ready) {
-            ready.set(isReady);
-            ready.notifyAll();
+    private static class PendingChanges implements Serializable {
```

Comment:
OK -- do you think that's something that would rise to the level of including it in the public API: https://docs.gradle.org/current/javadoc/org/gradle/BuildResult.html

e.g., as part of getting the `buildFinished` event, we would be able to tell if the build is already out-of-date (aka has pending changes)?